🛠 Tooling: Renovate config should run pnpm dedupe

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

I'm noticing that renovate PRs are failing on the `lint:packages` task:

* #394
* #418

https://github.com/JoshuaKGoldberg/template-typescript-node-package/actions/runs/4964139504/jobs/8884015134?pr=418:

```plaintext
 ERR_PNPM_DEDUPE_CHECK_ISSUES  Dedupe --check found changes to the lockfile
```

I see from https://docs.renovatebot.com/configuration-options/ that `pnpmDedupe` is listed as an option. Let's try that?

### Additional Info

_No response_

--- a/.github/renovate.json
+++ b/.github/renovate.json
@@ -3,5 +3,6 @@
 	"automerge": true,
 	"internalChecksFilter": "strict",
 	"labels": ["dependencies"],
+	"postUpdateOptions": ["pnpmDedupe"],
 	"stabilityDays": 3
 }

